arch/arm/src/stm32h7/hardware/stm32_fmc.h

```diff
@@ -0,0 +1,390 @@
+#define STM32_FMC_PATT2          (STM32_FMC_BASE+STM32_FMC_PATT2_OFFSET )
+#define STM32_FMC_PATT3          (STM32_FMC_BASE+STM32_FMC_PATT3_OFFSET )
+#define STM32_FMC_PATT4          (STM32_FMC_BASE+STM32_FMC_PATT4_OFFSET )
+
+#define STM32_PIO4                (STM32_FMC_BASE+STM32_FMC_PIO4_OFFSET )
+
+#define STM32_FMC_ECCR(n)        (STM32_FMC_BASE+STM32_FMC_ECCR_OFFSET(n))
+#define STM32_FMC_ECCR2          (STM32_FMC_BASE+STM32_FMC_ECCR2_OFFSET )
+#define STM32_FMC_ECCR3          (STM32_FMC_BASE+STM32_FMC_ECCR3_OFFSET )
+
+#define STM32_FMC_SDCR1          (STM32_FMC_BASE+STM32_FMC_SDCR1_OFFSET)
+#define STM32_FMC_SDCR2          (STM32_FMC_BASE+STM32_FMC_SDCR2_OFFSET)
+
+#define STM32_FMC_SDTR1          (STM32_FMC_BASE+STM32_FMC_SDTR1_OFFSET)
+#define STM32_FMC_SDTR2          (STM32_FMC_BASE+STM32_FMC_SDTR2_OFFSET)
+
+#define STM32_FMC_SDCMR          (STM32_FMC_BASE+STM32_FMC_SDCMR_OFFSET)
+#define STM32_FMC_SDRTR          (STM32_FMC_BASE+STM32_FMC_SDRTR_OFFSET)
+#define STM32_FMC_SDSR           (STM32_FMC_BASE+STM32_FMC_SDSR_OFFSET)
+
+/* Register Bitfield Definitions ****************************************************/
+
+#define FMC_BCR_MBKEN            (1 << 0)   /* Memory bank enable bit */
+#define FMC_BCR_MUXEN            (1 << 1)   /* Address/data multiplexing enable bit */
+#define FMC_BCR_MTYP_SHIFT       (2)        /* Memory type */
+#define FMC_BCR_MTYP_MASK        (3 << FMC_BCR_MTYP_SHIFT)
+#  define FMC_BCR_SRAM           (0 << FMC_BCR_MTYP_SHIFT)
+#  define FMC_BCR_ROM            (0 << FMC_BCR_MTYP_SHIFT)
+#  define FMC_BCR_PSRAM          (1 << FMC_BCR_MTYP_SHIFT)
+#  define FMC_BCR_CRAM           (1 << FMC_BCR_MTYP_SHIFT)
+#  define FMC_BCR_NOR            (2 << FMC_BCR_MTYP_SHIFT)
+#define FMC_BCR_MWID_SHIFT       (4)        /* Memory data bus width */
+#define FMC_BCR_MWID_MASK        (3 <<  FMC_BCR_MWID_SHIFT)
+#  define FMC_BCR_MWID8          (0 << FMC_BCR_MWID_SHIFT)
+#  define FMC_BCR_MWID16         (1 << FMC_BCR_MWID_SHIFT)
+#define FMC_BCR_FACCEN           (1 << 6)   /* Flash access enable */
+#define FMC_BCR_BURSTEN          (1 << 8)   /* Burst enable bit */
+#define FMC_BCR_WAITPOL          (1 << 9)   /* Wait signal polarity bit */
+#define FMC_BCR_WRAPMOD          (1 << 10)  /* Wrapped burst mode support */
+#define FMC_BCR_WAITCFG          (1 << 11)  /* Wait timing configuration */
+#define FMC_BCR_WREN             (1 << 12)  /* Write enable bit */
+#define FMC_BCR_WAITEN           (1 << 13)  /* Wait enable bit */
+#define FMC_BCR_EXTMOD           (1 << 14)  /* Extended mode enable */
+#define FMC_BCR_ASYNCWAIT        (1 << 15)  /* Wait signal during asynchronous transfers */
+#define FMC_BCR_CBURSTRW         (1 << 19)  /* Write burst enable */
+#define FMC_BCR_BMAP_SHIFT       (24)
+#  define FMC_BCR_BMAP_0               (0 << FMC_BCR_BMAP_SHIFT)
+#  define FMC_BCR_BMAP_1               (1 << FMC_BCR_BMAP_SHIFT)
+#  define FMC_BCR_BMAP_2               (2 << FMC_BCR_BMAP_SHIFT)
```

Comment:
bad alignment here too.
Sorry, in previous comment I should select multiple lines for comment.

Reply:
No, that's my fault.
I should check before push